content/components/dialog.mdx

```diff
@@ -0,0 +1,265 @@
+
+The `height` and `width` may be set to `auto` which will set a minimum width of `192px` and respond to the contents of the Dialog. Additionally, an explicit `height` and `width` may be set with a specific set of options.
+
+<img width="1292" alt="" src="https://user-images.githubusercontent.com/18661030/185178528-e1c176ef-f407-40ab-8729-965e21d64f6d.png"/>
+<Caption>Recommended height and width combinations for Dialog</Caption>
+
+### Spacing
+
+The edges of Dialog are free of text and heavy visuals to visually distinguish the Dialog as an elevated surface, alongside its shadow. Content within the body and header should remain within the 16px safe area. However, inner components that have transparent or discrete backgrounds may bleed into the padding for visual alignment.
+
+<img alt="" src="https://user-images.githubusercontent.com/293280/174925634-603a27cf-2281-4a3a-9078-d1b8138e0d09.png" width="960" />
+<Caption>
+  1. A search field in the Subheader region appears closer to the edges. While the search icon remains within the safe area.<br />
+  2. Close button placed `8px` away from the edges. Note that the "×" glyph remains within the safe area.<br />
+  3. Action list items in the Body region.<br />
+  4. Footer buttons always appear inside the safe area. <br />
+</Caption>
+
+## Usage
+
+### Keep content simple
+
+Dialogs are meant to be interacted with in a single context. Avoid creating a whole page inside a Dialog. Prefer single-column layouts that convey a clear goal from the moment the Dialog is opened.
+
+However, Dialogs are powerful tools when creating space for single-context interfaces. When designing a new feature to support a main interface, start by considering if the new feature can be created under a single context.
+
+If the answer is yes, designing the feature inside a Dialog provides an easy way to add responsive support, and avoid deeper navigations. If the answer is no, or if a side navigation is required for the feature, consider designing a new page.
+
+<DoDontContainer>
+  <Do>
+    <img width="464" alt="" src="https://user-images.githubusercontent.com/18661030/185224257-a40112ff-aaf3-4947-acf0-6b4020e023df.png" />
+    <Caption>Prefer a single column layout.</Caption>
+  </Do>
+  <Dont>
+    <img width="464" alt="" src="https://user-images.githubusercontent.com/18661030/185224316-0680b89a-8514-4336-9147-64aac722cd9a.png" />
+    <Caption>Avoid side navigation within a Dialog.</Caption>
+  </Dont>
+</DoDontContainer>
+
+### Trigger elements and keyboard focus
+
+Dialogs are initially hidden, and can be opened by an element with `role="button"`, called a **trigger**. The role `button` allows the **trigger** to be a relativelly complex element, such as an action list item.
+
+Dialogs should be fully functional using the keyboard and assistive technology. Pressing `Esc` should dismiss a Dialog, also returning focus to the **trigger** that opened it.
```

Comment:
This `should` language makes it seem optional vs. using `must`. Thoughts on adjusting?